landscaper/gardenlet/pkg/apis/imports/validation/validation.go

```diff
@@ -0,0 +1,138 @@
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+
+	"github.com/gardener/gardener/landscaper/gardenlet/pkg/apis/imports"
+	gardencore "github.com/gardener/gardener/pkg/apis/core"
+	gardenletconfig "github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	confighelper "github.com/gardener/gardener/pkg/gardenlet/apis/config/helper"
+	gardenletvalidation "github.com/gardener/gardener/pkg/gardenlet/apis/config/validation"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateLandscaperImports validates a Imports object.
+func ValidateLandscaperImports(imports *imports.Imports) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if imports.GardenCluster.Spec.Configuration.RawMessage == nil {
+		return append(allErrs, field.Required(field.NewPath("gardenCluster"), "the garden cluster kubeconfig has to be provided."))
+	}
+
+	if imports.SeedCluster.Spec.Configuration.RawMessage == nil {
+		return append(allErrs, field.Required(field.NewPath("seedCluster"), "the seed cluster kubeconfig has to be provided."))
+	}
+
+	if imports.DeploymentConfiguration != nil {
+		allErrs = append(allErrs, validateGardenletDeployment(imports.DeploymentConfiguration, field.NewPath("deploymentConfiguration"))...)
+	}
+
+	componentConfigurationPath := field.NewPath("componentConfiguration")
+
+	// Convert gardenlet config to an internal version
+	gardenletConfig, err := confighelper.ConvertGardenletConfiguration(imports.ComponentConfiguration)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(componentConfigurationPath, imports.ComponentConfiguration, fmt.Sprintf("could not convert to gardenlet configuration: %v", err)))
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validateGardenletConfiguration(gardenletConfig, componentConfigurationPath)...)
+
+	// if only the Seed specifies a backup configuration (componentConfiguration.SeedConfig.Spec.Backup) but not imports.SeedBackup
+	// then we assume that the reference backup secret already exists in the Garden cluster and does not have to
+	// be deployed by the landscaper. Hence, nothing to validate.
+	if imports.SeedBackup != nil {
+		allErrs = validateBackup(imports.SeedBackup, gardenletConfig.SeedConfig.Spec.Backup, componentConfigurationPath.Child("seedConfig.spec.backup"))
+	}
+
+	return allErrs
+}
+
+// validateGardenletConfiguration validates the Gardenlet configuration using the validation from the
+// Gardenlet package
+// in addition, validates settings specific to the Gardenlet landscaper
+func validateGardenletConfiguration(gardenletConfig *gardenletconfig.GardenletConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if gardenletConfig.SeedSelector != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedSelector"), "seed selector is forbidden. Provide a seedConfig instead."))
```

Comment:
Perhaps it's better to avoid multiple sentences in error messages? Such a style is rarely used, and doesn't read naturally from my perspective.
```suggestion
		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedSelector"), "seed selector is forbidden, provide a seedConfig instead."))
```